src/app/data-model/Nullable.h

```diff
@@ -37,29 +37,37 @@ inline constexpr auto NullNullable = NullOptional;
  * things.
  */
 template <typename T>
-struct Nullable : protected Optional<T>
+struct Nullable : protected std::optional<T>
 {
+
     //
     // The following 'using' statement is needed to make visible
     // all constructors of the base class within this derived class.
     //
-    using Optional<T>::Optional;
-
-    // Pull in APIs that make sense on Nullable with the same names as on
-    // Optional.
-    using Optional<T>::Value;
-    using Optional<T>::ValueOr;
+    using std::optional<T>::optional;
+    using std::optional<T>::value_or;
+    using std::optional<T>::value;
+    using std::optional<T>::has_value;
+    using std::optional<T>::operator*;
+    using std::optional<T>::operator->;
+
+    // backwards compatibiltiyt with old chip::Optional functionality
+    // NOTE: as we transition to std::optional, these should be removed
```

Comment:
No, I don't think they in fact should be.  We want to be very clear on where things are optional and where they are nullable, especially for things that are both optional and nullable.

This is why Nullable exists at all as a separate type.

Reply:
I think we should keep `NullOptionalType` hnowever for the rest I think we should replace `Value` with `value` and `ValueOr` with `value_or` as they seem to make sense and this is what we did with optional as well.

I will move the constructor to not be attached to this comment.

Reply:
> and this is what we did with optional as well

We did it with optional because we want to be API-compatible with std::optional there, with an eye to replacing it, yes?

We are not planning to replace Nullable with std::optional.  We shouldn't churn things just to make it look like std::optional when they are not in fact the same thing... and if it's a Matter-defined type, its API should look like other Matter types.

Reply:
The main difficulty will be that we will change a lot of files, so I expect a sed of `%s/ValueOr/value_or` and if the API surface between nullable and optional differs, that replacement will be a pain to only apply to optional but not nullable. 

We can defer that to the chip::Optional replacement though. PR is now updated.